src/components/ListLayout.jsx

```diff
@@ -61,6 +61,94 @@ const ListLayout = ({ items, localToken }) => {
     return timeCheck;
   }
 
+  // updates isActive property of item to true if item has 2+ purchases and has been purchased within calculated estimate
+  // isActive is defaulted to false when item is added
+  items.forEach((item) => {
+    const dateOfLastTransaction =
+      item.totalPurchases > 0 ? item.purchasedDate : item.createdAt;
+    const daysSinceLastTransaction =
+      (currentTime - dateOfLastTransaction) / oneDay;
+    if (
+      item.totalPurchases > 1 &&
+      daysSinceLastTransaction < 2 * item.previousEstimate
+    ) {
+      let itemToUpdate = {
+        ...item,
+        isActive: true,
+      };
+      setUpdateToDb(localToken, item.id, itemToUpdate);
+    }
+  });
+
+  //this sorts by previousEstimate (calculation of when user will buy the item again) Note: items is already sorted alphabetically by item Id which is the normalized item name
+  items
+    .sort((itemA, itemB) => itemA.previousEstimate - itemB.previousEstimate)
+    .sort((itemA, itemB) => Number(itemB.isActive) - Number(itemA.isActive));
+  let frequencyGroup;
+  function colorClass(item) {
+    //inactive = gray
+    if (!item.isActive) {
+      frequencyGroup = 'inactive';
+      return 'bg-gray-300';
+    }
+
+    //buy soon
+    else if (item.previousEstimate < 7) {
+      frequencyGroup = 'buy soon';
+      return 'bg-green-500';
+    }
+    //kind of soon
+    else if (item.previousEstimate < 30) {
+      frequencyGroup = 'kind of soon';
+      return '';
+    }
+    //not so soon
+    else {
+      frequencyGroup = 'not so soon';
+      return 'bg-rose-300';
+    }
+  }
+
+  const filteredItems = items.filter((item) =>
```

Comment:
@alenamedved great point! Especially since there's multiple places where filtering code will run: filteredItemsByString (or filteredItemsBySearchString) or such